pwndbg/commands/ipython_interactive.py

```diff
@@ -0,0 +1,42 @@
+"""
+Command to start an interactive IPython prompt.
+"""
+import sys
+from contextlib import contextmanager
+
+import gdb
+
+import pwndbg.commands
+
+
+@contextmanager
+def switch_to_ipython_env():
+    """We need to change stdout/stderr to the default ones, otherwise we can't use tab or autocomplete"""
+    # Save GDB's stdout and stderr
+    saved_stdout = sys.stdout
+    saved_stderr = sys.stderr
+    # Use Python's default stdout and stderr
+    sys.stdout = sys.__stdout__
+    sys.stderr = sys.__stderr__
+    yield
+    # Restore GDB's stdout and stderr
+    sys.stdout = saved_stdout
+    sys.stderr = saved_stderr
+    # Restore Python's default ps1 and ps2 for GDB's `pi` command
+    sys.ps1 = ">>> "
+    sys.ps2 = "... "
+
+
+@pwndbg.commands.ArgparsedCommand("Start an interactive IPython prompt.")
+def ipi():
+    with switch_to_ipython_env():
+        # Use `gdb.execute` to embed IPython into GDB's variable scope
+        code4ipython = """import IPython
+import jedi
+import pwn
+jedi.Interpreter._allow_descriptor_getattr_default = False
```

Comment:
What is `jedi`?

Also @Arusekk do u know if `pwn` import influences anything with the environment we are in? AFAIK pwntools suggested to use `pwnlib` in libraries using pwntools and `pwn` only for small scripts or interactive shells?

Reply:
> What is `jedi`?

`IPython` used `jedi` to build some features about autocomplete, and it is also related to the `@property` issue.

So I guess that's why we can affect `IPython` by `jedi.Interpreter._allow_descriptor_getattr_default`.